File: Competition/2019/Blue/Sentence-Maker/Solution.java
/* IPC Problem 7: Sentence Maker

 a) Process data from STDIN
 b) Check if the sentence can be made from the characters given
 b) Return output to the console
*/

import java.util.Scanner;
import java.util.ArrayList;

public class Solution
{
    public static void main(String[] args)
    {
        // String to check if making sentence is possible
        String possible = "";

        // Get input from STDIN
        Scanner myScanner = new Scanner(System.in);

        System.out.println("Enter the list of letters:");
        String chars = myScanner.nextLine();

        System.out.println("Please enter the sentence to be formed:");
        String sentence = myScanner.nextLine();

        myScanner.close();

        // Get ArrayLists of characters out of the strings given by the user
        ArrayList<Character> allChars = getCharArrFromStr(chars);
        ArrayList<Character> sentenceChars = getCharArrFromStr(sentence);

        // Remove all characters in the sentence Chars from the allChars array
        for (Character theChar : sentenceChars)
        {
            if (allChars.contains(theChar))
            {
                allChars.remove(theChar);
            }

            else
            {
                possible = "not possible";
                break;
            }
        }

        if (possible.length() == 0)
        {
            possible = "possible";
        }

        System.out.println(possible);
    }

    public static ArrayList<Character> getCharArrFromStr(String myStr)
    {
        ArrayList<Character> charArr = new ArrayList<Character>();

        for (char theChar : myStr.toCharArray())
        {
            if (Character.isAlphabetic(theChar))
            {
                charArr.add(theChar);
            }
        }

        return charArr;
    }
}

File: Competition/2019/Blue/Sentence-Maker/solution.py
# IPC Problem 7: Sentence Maker

# a) Process data from STDIN
# b) Check if the sentence can be made from the characters given
# b) Return output to the console

def main():

    # Check if it is possible to form sentence
    possible = ""
    
    # Get data from STDIN
    chars, sentence = getSTDIN()

    # Process each as a character array
    char_arr_for_chars = get_char_list(chars)
    char_arr_for_sentence = get_char_list(sentence)

    for char in char_arr_for_sentence:

        try:
            char_arr_for_chars.remove(char)
        
        except:
            possible = "not possible"
            break

    if len(possible) == 0:
        possible = "possible"
    
    print(possible)
    

def getSTDIN():
    """Get input from STDIN."""

    print("Enter the list of letters:")
    chars = input()

    print("Please enter the sentence to be formed:")
    sentence = input()

    return chars, sentence

def get_char_list(string):
    """Returns a list of all alphabetic characters from a string"""

    charlist = []

    for char in string:

        if char.isalpha():
            charlist.append(char)
    
    return charlist

main()